include/nadjieb/mjpeg_streamer.hpp

```diff
@@ -258,6 +268,14 @@ class MJPEGStreamer {
                         this->master_socket_, reinterpret_cast<struct sockaddr*>(&(this->address_)),
                         reinterpret_cast<socklen_t*>(&addrlen));
                     this->panicIfUnexpected(new_socket < 0, "ERROR: accept\n");
+                    auto remapped = --this->accept_socket_current_;
+                    this->panicIfUnexpected(
+                        dup2(new_socket, remapped) < 0, "ERROR: remap socket\n");
+                    if (remapped < this->accept_socket_start_ - this->accept_socket_max_) {
```

Comment:
Would be a problem *if* threading was required ( I don't think it is).
https://en.wikipedia.org/wiki/Compare-and-swap
https://en.cppreference.com/w/cpp/atomic/atomic/compare_exchange